### PR TITLE
chore: Add SymbolUsageKind to GraphQL API

### PR DESCRIPTION
**Pull request by varungandhi-src**
Mostly a documentation-related PR, the implementation will be done later.

Fixes https://linear.app/sourcegraph/issue/GRAPH-697/add-symbolusagekind-to-usages

## Test plan

n/a

## Changelog

### CODE REVIEW

File: cmd/frontend/graphqlbackend/codeintel.codenav.graphql [hunk 1]
```diff
@@ -721,6 +721,149 @@ type Usage {
...
+    """
+    usageKind: SymbolUsageKind!
+}
+
+"""
+Categorizes a usage based on its relationship to the symbol of interest.
+
+This enum may be expanded in the future.
+
+EXPERIMENTAL: This type may make backwards-incompatible changes in the future.
```
Comment (keynmol) [inline cmd/frontend/graphqlbackend/codeintel.codenav.graphql:737]:
```suggestion
EXPERIMENTAL: This type may change in a backwards-incompatible way in the future.
```

File: cmd/frontend/graphqlbackend/codeintel.codenav.graphql [hunk 2]
```diff
@@ -721,6 +721,149 @@ type Usage {
...
+    For example, consider the following pseudocode:
+
+    interface Animal:
+        sound()
+
+    class Dog implements Animal:
+        sound() override { ... }
+
+    func bark(dog: Dog):
+                   ^-^ (1)
```
Comment (keynmol) [inline cmd/frontend/graphqlbackend/codeintel.codenav.graphql:839]: kawaii

**Review by keynmol — APPROVED** [review]
Really fantastic docs 👍

File: cmd/frontend/graphqlbackend/codeintel.codenav.graphql [hunk 3]
```diff
@@ -721,6 +721,149 @@ type Usage {
...
+        sound()
+
+    class Dog implements Animal:
+        sound() override { ... }
+
+    func makeSounds(animal: Animal, dog: Dog):
+        animal.sound()
+        #      ^---^ (1)
+        dog.sound()
+        #   ^---^ (2)
```
Comment (kritzcreek) [inline cmd/frontend/graphqlbackend/codeintel.codenav.graphql:753]: Did you mean to move these markers up into the `interface` and `class` blocks?

Comment (varungandhi-src) [reply cmd/frontend/graphqlbackend/codeintel.codenav.graphql:753]: No, the (1) and (2) mark the range passed as input. When I say "usageForSymbol for (1)" I'm talking about the input range. Do you think I should add more markers and clarify or change the wording?

Comment (kritzcreek) [reply cmd/frontend/graphqlbackend/codeintel.codenav.graphql:753]: The thing that tripped me up is that this is the `SymbolUsageKind` for definitions, and both of these markers pointed at references.
I'll have to read this again on Monday, maybe I'm misunderstanding what this is about.

Comment (varungandhi-src) [reply cmd/frontend/graphqlbackend/codeintel.codenav.graphql:753]: Right, so the output ranges will correspond to the declarations, but the input range may correspond to any site.

**Pull request merged**